x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java

```diff
@@ -169,21 +177,25 @@ private void inferAgainstAllocatedModel(Request request, Response.Builder respon
     }
 
     private void inferSingleDocAgainstAllocatedModel(
+        Task task,
         String modelId,
         InferenceConfigUpdate inferenceConfigUpdate,
         Map<String, Object> doc,
         ActionListener<InferenceResults> listener
     ) {
+        TaskId taskId = new TaskId(clusterService.localNode().getId(), task.getId());
+        InferTrainedModelDeploymentAction.Request request = new InferTrainedModelDeploymentAction.Request(
+            modelId,
+            inferenceConfigUpdate,
+            Collections.singletonList(doc),
+            TimeValue.MAX_VALUE
+        );
+        request.setParentTaskId(taskId);
         executeAsyncWithOrigin(
-            client,
+            new ParentTaskAssigningClient(client, taskId),
```

Comment:
`ParentTaskAssigningClient` calls `request.setParentTaskId(taskId);` exactly as you have done in line 193 you should either set `request.setParentTaskId(taskId);` yourself or use the `ParentTaskAssigningClient` not both.

https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/client/ParentTaskAssigningClient.java#L54

Reply:
I tried that at first. When the internal infer calls the trained model infer action, two tasks are created for the trained model infer action. One tagged with `[n]` for network and another one. If I don't use a `ParentTaskAssigningClient` the network task is not parented properly.

Reply:
Actually, I revisited this. From testing it seems that setting the parent task id on the request has no effect. Only when calling with the `ParentTaskAssigningClient` the result task is parented correctly. I'll follow up with a PR to remove setting the parent task id on the request.

Reply:
I've now raised https://github.com/elastic/elasticsearch/pull/80905